include/highfive/h5easy_bits/H5Easy_Eigen.hpp

```diff
@@ -23,41 +23,36 @@ namespace eigen {
 
 // return the shape of the "Eigen::Matrix" as size 1 or 2 "std::vector<size_t>"
 template <class T>
-inline std::vector<size_t> shape(const T& data)
-{
-    if (std::decay_t<T>::RowsAtCompileTime == 1) {
+inline std::vector<size_t> shape(const T& data) {
+    if (std::decay<T>::type::RowsAtCompileTime == 1) {
         return {static_cast<size_t>(data.cols())};
     }
-
-    if (std::decay_t<T>::ColsAtCompileTime == 1) {
+    if (std::decay<T>::type::ColsAtCompileTime == 1) {
         return {static_cast<size_t>(data.rows())};
     }
-
     return {static_cast<size_t>(data.rows()),
             static_cast<size_t>(data.cols())};
 }
 
+
+using EigenIndex = Eigen::DenseIndex;
```

Comment:
Just out of curiosity: what is the difference between `Eigen::Index` and `Eigen::DenseIndex`?

Reply:
Index is apparently generic, but became public API only in 3.3. From my understanding there are indexes for Sparse Matrixes.